app/experimenter/docs/adrs/0008-integrating-htmx.md

```diff
@@ -0,0 +1,79 @@
+- Date: 2022-10-25
+
+## Context and Problem Statement
+
+Nimbus UI uses GraphQL to fetch data from the backend and it highly depends on the three packages i.e. [graphene-django](https://github.com/graphql-python/graphene-django), [graphene](https://github.com/graphql-python/graphene) and [apollo-client](https://github.com/apollographql/apollo-client) and we have seen these packages not getting released new versions for a while. We have found out that using GraphQL significantly hinders the performance of the Experimenter as it adds extra overhead while making the call and the benefit of using GraphQL is not something we are making use of. We also have to significantly maintain the burden to implement most of the API in both DRF (Django Rest Framework) and GraphQL, and additional testing and tools are required to support that. As our application is growing so what steps we should take to improve the performance and better maintainability of the application?
+
+## Decision Drivers
+
+- Improve performance which leads to better UX Experience
+- Reduce dependency on extra packages
+- Better development tooling
+- Better maintainability of the application
+
+## Considered Options
+
+- Get rid of GraphQL and transition to REST Api's and keeping the Typescript/React frontend
+- Designing new pages with HTMX, and keep the existing UI (React and GraphQL) and eventually move to HTMX
+
+## Decision Outcome
+
+Chosen option: After careful consideration, we have decided to choose the option - Designing new pages with HTMX, and keep the existing UI (React and GraphQL), and eventually moving to HTMX. Moving to HTMX is simpler in terms of backend compatibility as we can use our existing code without any major modifications. Work includes creating new pages using HTMX, setting up test cases to test HTMX pages, and evaluating the success/failure of HTMX on the Experimenter platform, so that if we are successful we can start the transition from React to HTMX completely. Whereas in option 1 we need to actively work on the backend and frontend to transit from GraphQL to REST Apis. This can only work once we have converted everything to Rest Api's i.e. both frontend and backend. Particular V5 API of the experimenter is not used anywhere else apart from the front end, so if we switch to HTMX, we will be not dependent on any kind of APIs and will be closer to the server. The main benefit of choosing option 2 is that it can co-exist with React/GraphQL, meanwhile, we will update to the latest dependency packages to improve performance and can slowly progress towards 100% HTMX.
+
+### Positive Consequences
+
+- All web pages using HTMX!
+- Better performance
+- Learn new tool HTMX
+
+### Negative Consequences
+
+- We'd be looking for to determine whether the new pages aren't successful. The one possible solution would be to re-write those pages in Typescript/React and instead of GraphQL and transition to Rest API.
```

Comment:
Possibly out of scope for this document, but do we have a plan for determining success / evaluating performance vs existing pages, especially given that we won't have a 1:1 comparison (new pages will be HTMX and old pages won't be re-written at first)?

If this is out of scope, please ignore for now and it can be discussed in a different forum.

Reply:
this document doesn't address that, but I will write separate doc for that :)